### DISCUSSION TITLE
Undefined symbols for architecture arm64

### DISCUSSION THREAD

**Discussion by aliesh1996** [Z3Prover/z3#6904] [Q&A]
I always get this error on Mac OS M1 (arm64) whenever I try to compile z3++.h:

g++ -arch arm64 -std=gnu++11 -o test test.cpp -I/opt/homebrew/Cellar/z3/4.12.2/include
Undefined symbols for architecture arm64:
  "_Z3_del_config", referenced from:
      z3::config::~config() in test-acd1c5.o
  "_Z3_del_context", referenced from:
      z3::context::~context() in test-acd1c5.o
  "_Z3_mk_config", referenced from:
      z3::config::config() in test-acd1c5.o
  "_Z3_mk_context_rc", referenced from:
      z3::context::init(z3::config&) in test-acd1c5.o
  "_Z3_set_ast_print_mode", referenced from:
      z3::context::set_context(_Z3_context*) in test-acd1c5.o
  "_Z3_set_error_handler", referenced from:
      z3::context::set_context(_Z3_context*) in test-acd1c5.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

I tried many ways to resolve this, but nothing worked. Can someone help me how to configure z3 on Mac M1 to compile and run z3++.h successfully?

**Comment by wintersteiger** [comment]
Perhaps missing a `-L <directory containing libz3.dylib>` and a `-lz3`?
